wom/src/main/scala/wom/expression/InputPointerToWomValue.scala

```diff
@@ -21,14 +21,22 @@ object InputPointerToWomValue extends Poly1 {
   implicit def fromOutputPort: Case.Aux[OutputPort, ToWdlValueFn] = at[OutputPort] {
     port => (knownValues: Map[String, WomValue], ioFunctions: IoFunctionSet, outputPortLookup: OutputPortLookup, inputDefinition: InputDefinition) =>
       (outputPortLookup(port), inputDefinition) match {
-        case (v: Valid[WomValue], _) => v.map(inputDefinition.valueMapper(ioFunctions))
+        case (Valid(womValue), _) if isDefined(womValue) =>
+          inputDefinition.valueMapper(ioFunctions)(womValue).valid
         case (_, InputDefinitionWithDefault(_, _, defaultExpression, valueMapper)) =>
           evaluate(defaultExpression, knownValues, ioFunctions).map(valueMapper(ioFunctions))
         case (_, OptionalInputDefinition(_, optionalType, valueMapper)) => valueMapper(ioFunctions)(optionalType.none).validNel
         case _ => s"Failed to lookup input value for required input ${port.name}".invalidNel
       }
   }
 
+  private def isDefined(womValue: WomValue) = {
+    womValue match {
+      case WomOptionalValue(_, None) => false
+      case _ => true
```

Comment:
Not sure if possible in CWL but generally
`case WomOptionalValue(_, Some(inner: WomOptinalValue)) => isDefined(inner)`